### PR TITLE
Chore/minimal consensus info

### PR DESCRIPTION
**Pull request by mxmar**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

Fixes `[2021-07-27 11:59:08] ERROR vanguard-chain: Failed to receive minimalConsensusInfo error=rpc error: code = Unavailable desc = Could not get MinimalConsensusInfoRange for a stream: invalid validators len, expected: 31, got: 30, epoch: 0x0`. Issue in `ProposerAssignments`.

Fixes `rpc error: code = Internal desc = Could not retrieve any state for epoch 1`. Issue in `MinimalConsensusInfoRange`.

**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

### CODE REVIEW

**Comment by mxmar** [comment]
```
[2021-07-27 11:05:39] TRACE vanchain: Got new minimal consensus info context=awaiting to fetch minimalConsensusInfo from stream
[2021-07-27 11:05:39]  INFO event: registered new subscriber for consensus info alreadyKnown=[0xc0001c5860 0xc0001c5e90 0xc0003cc660 0xc0003cca50 0xc0003302d0 0xc0003306f0 0xc000330b10 0xc000330f60 0xc0003313b0 0xc000331800 0xc000331c20 0xc00a13c090 0xc00a13c4b0 0xc00a13c900 0xc00a13cd20 0xc00a13d170 0xc00a13d590 0xc00a13d9e0 0xc00a13de00 0xc00a23e270 0xc00a23e690 0xc00a23eae0 0xc00a23ef00 0xc00a23f380 0xc00a23f7a0 0xc00a23fbf0] fromEpoch=0
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=0 epochStartTime=1627372084
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=1 epochStartTime=1627372276
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=2 epochStartTime=1627372468
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=3 epochStartTime=1627372660
INFO [07-27|11:05:39.981] Received minimalConsensusInfo            epoch=0 epochTimeStart=1627372081 validatorListLen=32
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=4 epochStartTime=1627372852
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=5 epochStartTime=1627373044
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=6 epochStartTime=1627373236
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=7 epochStartTime=1627373428
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=8 epochStartTime=1627373620
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=9 epochStartTime=1627373812
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=10 epochStartTime=1627374004
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=11 epochStartTime=1627374196
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=12 epochStartTime=1627374388
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=13 epochStartTime=1627374580
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=14 epochStartTime=1627374772
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=15 epochStartTime=1627374964
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=16 epochStartTime=1627375156
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=17 epochStartTime=1627375348
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=18 epochStartTime=1627375540
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=19 epochStartTime=1627375732
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=20 epochStartTime=1627375924
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=21 epochStartTime=1627376116
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=22 epochStartTime=1627376308
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=23 epochStartTime=1627376500
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=24 epochStartTime=1627376692
[2021-07-27 11:05:39]  INFO event: sending already known consensus info to subscriber epoch=25 epochStartTime=1627376884
INFO [07-27|11:05:40.002] Received minimalConsensusInfo            epoch=1 epochTimeStart=1627372273 validatorListLen=32
INFO [07-27|11:05:40.017] Received minimalConsensusInfo            epoch=2 epochTimeStart=1627372465 validatorListLen=32
INFO [07-27|11:05:40.025] Received minimalConsensusInfo            epoch=3 epochTimeStart=1627372657 validatorListLen=32
INFO [07-27|11:05:40.031] Received minimalConsensusInfo            epoch=4 epochTimeStart=1627372849 validatorListLen=32
INFO [07-27|11:05:40.036] Received minimalConsensusInfo            epoch=5 epochTimeStart=1627373041 validatorListLen=32
INFO [07-27|11:05:40.040] Received minimalConsensusInfo            epoch=6 epochTimeStart=1627373233 validatorListLen=32
INFO [07-27|11:05:40.044] Received minimalConsensusInfo            epoch=7 epochTimeStart=1627373425 validatorListLen=32
INFO [07-27|11:05:40.048] Received minimalConsensusInfo            epoch=8 epochTimeStart=1627373617 validatorListLen=32
INFO [07-27|11:05:40.052] Received minimalConsensusInfo            epoch=9 epochTimeStart=1627373809 validatorListLen=32
INFO [07-27|11:05:40.055] Received minimalConsensusInfo            epoch=10 epochTimeStart=1627374001 validatorListLen=32
INFO [07-27|11:05:40.059] Received minimalConsensusInfo            epoch=11 epochTimeStart=1627374193 validatorListLen=32
INFO [07-27|11:05:40.062] Received minimalConsensusInfo            epoch=12 epochTimeStart=1627374385 validatorListLen=32
INFO [07-27|11:05:40.065] Received minimalConsensusInfo            epoch=13 epochTimeStart=1627374577 validatorListLen=32
INFO [07-27|11:05:40.069] Received minimalConsensusInfo            epoch=14 epochTimeStart=1627374769 validatorListLen=32
INFO [07-27|11:05:40.072] Received minimalConsensusInfo            epoch=15 epochTimeStart=1627374961 validatorListLen=32
INFO [07-27|11:05:40.075] Received minimalConsensusInfo            epoch=16 epochTimeStart=1627375153 validatorListLen=32
INFO [07-27|11:05:40.078] Received minimalConsensusInfo            epoch=17 epochTimeStart=1627375345 validatorListLen=32
INFO [07-27|11:05:40.081] Received minimalConsensusInfo            epoch=18 epochTimeStart=1627375537 validatorListLen=32
INFO [07-27|11:05:40.084] Received minimalConsensusInfo            epoch=19 epochTimeStart=1627375729 validatorListLen=32
INFO [07-27|11:05:40.087] Received minimalConsensusInfo            epoch=20 epochTimeStart=1627375921 validatorListLen=32
INFO [07-27|11:05:40.090] Received minimalConsensusInfo            epoch=21 epochTimeStart=1627376113 validatorListLen=32
INFO [07-27|11:05:40.092] Received minimalConsensusInfo            epoch=22 epochTimeStart=1627376305 validatorListLen=32
INFO [07-27|11:05:40.094] Received minimalConsensusInfo            epoch=23 epochTimeStart=1627376497 validatorListLen=32
INFO [07-27|11:05:40.097] Received minimalConsensusInfo            epoch=24 epochTimeStart=1627376689 validatorListLen=32
INFO [07-27|11:05:40.099] Received minimalConsensusInfo            epoch=25 epochTimeStart=1627376881 validatorListLen=32
```

It has passed all consensus info. Looks like it is working.

File: beacon-chain/core/helpers/committee.go [hunk 1]
```diff
@@ -440,14 +439,30 @@ func ProposerAssignments(
...
+		proposerSlots = append(proposerSlots, slot)
+		assign := &ethpb.ValidatorAssignments_CommitteeAssignment{
+			ProposerSlots:  proposerSlots,
+			PublicKey:      pubKey[:],
+			ValidatorIndex: i,
+		}
+
+		proposerIndexToSlots = append(proposerIndexToSlots, assign)
+
+		if len(proposerIndexToSlots) > sliceRange {
```
Comment (mxmar) [inline beacon-chain/core/helpers/committee.go:463]: This should never happen. But this might be some crazy edge case - all tests didn't notice this wrong behavior but I can't be 100% sure here, so I've manage to leave some condition for this and have clean proper log output to see what was the state (all variables etc.) when it occur.

**Comment by atif-konasl** [comment]
@mxmar do we need this PR now? We can close it.
